ethcore/src/client/client.rs

```diff
@@ -187,6 +186,20 @@ impl Client {
 			try!(db.write(batch).map_err(ClientError::Database));
 		}
 
+		trace!("Cleanup journal: DB Earliest = {:?}, Latest = {:?}", state_db.journal_db().earliest_era(), state_db.journal_db().latest_era());
+		if let (Some(earliest), Some(latest)) = (state_db.journal_db().earliest_era(), state_db.journal_db().latest_era()) {
+			if latest > earliest && latest - earliest > config.history {
+				let mut era = earliest;
+				while era <= latest - config.history {
+					trace!("Removing era {}", era);
+					let batch = DBTransaction::new(&db);
+					try!(state_db.journal_db_mut().commit_old(&batch, era, &chain.block_hash(era).expect("Old block not found in the database")));
```

Comment:
This is exactly why I made https://github.com/ethcore/parity/pull/2329. There is no need for `commit`, `commit_old` since they work the same way.

It should just `mark_canonical` in a loop.